libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -855,6 +856,15 @@ bool AP_Arming::mission_checks(bool report)
         check_failed(ARMING_CHECK_MISSION, report, "Failed to open %s", AP_MISSION_SDCARD_FILENAME);
     }
 #endif
+
+    // do not allow arming if there are no mission items and we are in
+    // (e.g.) AUTO mode
+    if (AP::vehicle()->current_mode_requires_mission() &&
```

Comment:
Should this be before the per item checks? You could also break out the nullptr check from 803 to give that more useful message in that case.